### PR TITLE
Feature/metric endpoint

### PR DESCRIPTION
**Pull request by CalvinRodo**
Adds an events endpoint to the submission server

This end point uses basic auth. 
It takes one parameter which is the date you would like to query for events 

You will get back a json array containing objects with the following properties:

 - identifier - The event that occurred
 - source - The source of the event (PT, Bearer Token)
 - date - The date the events occurred on
 - count - The number of time the event occurred in a day

I also did some refactoring on the TestNewKeyClaim method to make it easier to identify where it has failed as well as the state of the unit being tested. 

The submission server requires the following Environment Variables to be present 

METRICS_USERNAME
METRICS_PASSWORD

Mocks were also updated. 

You can test this locally by running the following curl command which will return all events on 2020-10-05. 

This assumes that you are running in the dev container and haven't modified the METRICS_PASSWORD and METRICS_USERNAME 

`curl -POST --user "foo:bar" "http://127.0.0.1:8001/events/2020-10-05"`

### CODE REVIEW

File: pkg/server/metrics.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+func authorizeRequest(r *http.Request) error {
+
+	uname, pword, ok := r.BasicAuth()
+	if !ok {
+		return fmt.Errorf("basic auth required for access")
+	}
+
+	metricUsername, uok := os.LookupEnv("METRICS_USERNAME")
```
Comment (maxneuvians) [inline pkg/server/metrics.go:45]: You can probably just re-assign to `ok` because the panic will terminate the flow.

File: pkg/config/config.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ type Constants struct {
 	DisableCurrentDateCheckFeatureFlag bool
 	EnableEntirePeriodBundle           bool
 	RegionCode                         string
+	EventQueryRangeDates int
```
Comment (maxneuvians) [inline pkg/config/config.go:29]: fmt spacing

File: pkg/server/metrics.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	if !uok {
+		log(nil, nil).Panic("Metrics username not set")
```
Comment (maxneuvians) [inline pkg/server/metrics.go:47]: Does log.Panic also terminate flow? In other places the code terminates the service if it is missing https://github.com/cds-snc/covid-alert-server/blob/master/pkg/keyclaim/authenticator.go#L25

**Review by maxneuvians — APPROVED** [review]
LGTM, just some small syntactic changes

File: pkg/server/metrics.go (hunk 3 above)
Comment (CalvinRodo) [reply pkg/server/metrics.go:47]: Now checking this on app start

File: pkg/config/config.go (hunk 2 above)
Comment (CalvinRodo) [reply pkg/config/config.go:29]: go fmt ftw!

File: pkg/server/metrics.go (hunk 1 above)
Comment (CalvinRodo) [reply pkg/server/metrics.go:45]: moved the check to app.go

**Review by maxneuvians — APPROVED** [review]

**Pull request merged**